src/sage/matrix/special.py

```diff
@@ -2407,6 +2407,44 @@ def companion_matrix(poly, format='right'):
     return M
 
 
+def random_subset(n, k):
+    """
+    Return a random subset of range(n) of size k.
+
+    The distribution law is uniform.
+
+    EXAMPLES::
+
+        sage: from sage.matrix.special import random_subset
+        sage: random_subset(14,2)  # random
+        [5, 6]
+        sage: random_subset(14,8)  # random
+        [0, 3, 5, 6, 8, 10, 11, 12]
+
+    TESTS::
+
+        sage: random_subset(14,0)  # random
+        []
+        sage: random_subset(14,14)  # random
+        [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
+    """
+    if 2 * k > n:
+        use_complement = True
+        k = n - k
+    else:
+        use_complement = False
+
+    subset = []
+    available = list(range(n))
```

Comment:
It should be better to shuffle the list `available` and then to return the first `k` items (or complement `n - k` items).

Alternatively, if you prefer successive calls to `randint`, then instead of calling `available.pop(index)`, you can simply do `available[index] = available[-1]`.